### PR TITLE
Provide upstream exception when column conversion fails

### PR DESCRIPTION
**Pull request by taldcroft**
Partial fix for #4749.

### CODE REVIEW

**Comment by pllim** [comment]
Perhaps add a change log?

**Comment by taldcroft** [comment]
@pllim - yup, just waiting for tests.  It's always annoying that one cannot do the change log on the first commit because the PR number is not known.

**Comment by pllim** [comment]
@taldcroft , I usually take my chances by adding one to the current highest issue/PR :smile:

**Comment by astrofrog** [comment]
@taldcroft @pllim - we should make a script that can give us that number from the command-line :)

**Comment by taldcroft** [comment]
Both fine solutions.  Now I need to get my head around context localization in Python 3 that is making tests fail.

File: astropy/io/ascii/tests/test_read.py [hunk 1]
```diff
@@ -1107,3 +1107,22 @@ def test_path_object():
...
+    conversion error is provided.
+    """
+    ipac = """\
+| col0   |
+| double |
+ 1  2
+"""
+    with pytest.raises(ValueError) as err:
+        ascii.read(ipac, guess=False, format='ipac')
+    assert 'invalid literal for float()' in str(err.value)
```
Comment (astrofrog) [inline astropy/io/ascii/tests/test_read.py]: Either just use ``err.value.args[0]`` or move the assert inside the context manager

**Comment by taldcroft** [comment]
@astrofrog - the issue was with the context variable in exception handling.  This code behaves differently in Py2 and 3:
```
try:
    1/0
except Exception as err:
    pass
finally:
    # statement succeeds in Py2 but raises NameError (no err defined) in Py3
    print('finally ', err)
```
But in a context manager the context variable persists after the context block in both Py2 and 3.

**Comment by pllim** [comment]
>  a script that can give us that number from the command-line

@astrofrog , this script is more appropriate as a native git command. Perhaps we file an issue with GitHub?

**Comment by bsipocz** [comment]
> @astrofrog , this script is more appropriate as a native git command. Perhaps we file an issue with GitHub?

@pllim - and then it may or may not happen, and definitely not soon. (Not many (0) of my github feature requests got implemented, though all of them are brilliant ones ;) ) 

I think it's not very complicated to implement, and if that's right I try to add something to astropy-tools this evening.

**Comment by hamogu** [comment]
Looks good to me.

**Comment by astrofrog** [comment]
@taldcroft - could you let me know what to include for the changelog entry, or commit one directly to master?

**Comment by taldcroft** [comment]
```
- Provide more detail in the error message when reading a table fails due to a problem
  converting column string values. [#4759]
```

**Pull request merged**